### ISSUE TITLE
[Feature] Provide your own OAuth provider

### ISSUE THREAD

**Issue by kmcgill88** [bytedance/unpub#88]
I was wondering if there are any plans to add support for providing different OAuth providers. I see things default to Google now.

OAuth could be extracted to an abstract class similar to `MetaStore` and `PackageStore` and passed in.

If I took a stab at it would the project be open to merging?

**Comment by kmcgill88** [comment]
Made a draft pr to get some feedback. https://github.com/bytedance/unpub/pull/89

**Comment by talisk** [comment]
> I was wondering if there are any plans to add support for providing different OAuth providers. I see things default to Google now.
> 
> OAuth could be extracted to an abstract class similar to `MetaStore` and `PackageStore` and passed in.
> 
> If I took a stab at it would the project be open to merging?

Of course. If there's a stable custom auth provider support, please fell free to open PR. Welcome contributing.
